aptos-move/mvhashmap/src/versioned_group_data.rs

```diff
@@ -578,21 +317,36 @@ impl<
     /// modification otherwise). When consistent, the output is Ok(..).
     pub fn finalize_group(
         &self,
-        key: &K,
+        group_key: &K,
         txn_idx: TxnIndex,
     ) -> anyhow::Result<(Vec<(T, ValueWithLayout<V>)>, ResourceGroupSize)> {
-        let mut v = self.group_values.get_mut(key).expect("Path must exist");
-
-        v.commit_idx(ShiftedTxnIndex::new(txn_idx), false)?;
-        v.get_committed_group()
-    }
+        let superset_tags = self
+            .group_tags
+            .get(group_key)
+            .expect("Group tags must be set")
+            .clone();
```

Comment:
you should add a comment that superset_tags here is not deterministic - but it shouldn't matter, as below they will be filtered out deterministically